code/datums/actions/build_designator.dm

```diff
@@ -0,0 +1,243 @@
+
+	hologram.setDir(new_dir)
+	hologram.remove_filter("invalid_turf_filter")
+
+	if(!check_turf_validity(target_turf))
+		hologram.add_filter("invalid_turf_filter", 1, color_matrix_filter(rgb(233, 23, 23)))
+		return FALSE
+	return TRUE
+
+/// check if the turf is valid or not for the selected build type, and apply a matrix color if not
+/datum/action/ability/activable/build_designator/proc/check_turf_validity(turf/open/target_turf)
+	if(!isopenturf(target_turf))
+		return FALSE
+	if(!target_turf.allow_construction)
+		return FALSE
+	var/area/area = get_area(target_turf)
+	if(area.area_flags & NO_CONSTRUCTION)
+		return FALSE
+	if(isspaceturf(target_turf))
+		return FALSE
+	var/border_obj = construct_type::atom_flags & ON_BORDER
+	for(var/obj/object in target_turf)
+		if(!object.density)
+			continue
+		if(border_obj && (object.atom_flags & ON_BORDER) && !(object.dir & hologram.dir))
+			continue
+		return FALSE
+	return TRUE
+
+///Sets the cleanup timer
+/datum/action/ability/activable/build_designator/proc/start_cleanup_timer()
+	delete_timer()
+	cleanup_timer = addtimer(CALLBACK(src, PROC_REF(cleanup_hologram)), cleanup_time, TIMER_STOPPABLE)
+
+///Clears the cleanup timer
+/datum/action/ability/activable/build_designator/proc/delete_timer()
+	deltimer(cleanup_timer)
+	cleanup_timer = null
+
+///Removes the hologram
+/datum/action/ability/activable/build_designator/proc/cleanup_hologram()
+	delete_timer()
+	QDEL_NULL(hologram)
+
+//The actual building hologram
+/obj/effect/build_designator
+	anchored = TRUE
+	layer = ABOVE_ALL_MOB_LAYER
+	hud_possible = list(ORDER_HUD)
+	///Material needed for the recipe
+	var/obj/material_type
+	///Recipe for what we are building
+	var/datum/stack_recipe/recipe
+	///The visual effect we're attaching
+	var/image/holder
+	///Mob that is currently trying to build the recipe
+	var/mob/builder
+
+/obj/effect/build_designator/Initialize(mapload, obj/construct_type, mob/builder) //construct_type is a TYPE but typecast for initial values below
+	if(!construct_type)
+		return INITIALIZE_HINT_QDEL
+
+	material_type = GLOB.designator_types[construct_type]
+	recipe = GLOB.stack_recipes[material_type][construct_type]
+
+	//Because we only want this hologram visible via the right hud, but we want all the nice effects, we fully configure the appearance,
+	//steal it for the hud image, then wipe the objects appearance entirely
+
+	dir = builder.dir
+	icon = construct_type::icon
+	icon_state = construct_type::icon_state
+	faction = builder.faction
+
+	name = "holo [construct_type::name]"
+	desc = "A holographic representation of a [construct_type::name]. Apply [recipe.req_amount] [material_type::name] to build it."
+	. = ..()
+	SEND_GLOBAL_SIGNAL(COMSIG_GLOB_HOLO_BUILD_INITIALIZED, src)
+	prepare_huds()
+	makeHologram(0.7)
+
+	holder = hud_list[ORDER_HUD]
+	holder.appearance = appearance
+	holder.dir = dir
+	holder.alpha = 190
+	hud_list[ORDER_HUD] = holder
+
+	icon = null
+	cut_overlays()
```

Comment:
can you do alpha = 255 instead here

Reply:
uh, you mean alpha = 0? Answer is no, it completely breaks it and causes a bunch of hilariously bizarre visual stuff